### PR TITLE
feat: github build status

### LINKED ISSUE #14: Push build status back to github

### PR DESCRIPTION
**Pull request by njoyard**
Uses github API to update build status on commits.

Closes #14

### CODE REVIEW

File: lib/status/status.js [hunk 1]
```diff
@@ -1,10 +1,47 @@
 const { resolve } = require('path')
 const { mkdir, readFile, writeFile } = require('fs-extra')
+const GitHub = require('github-api')
```
Comment (xcambar) [inline lib/status/status.js]: Why not use the official Github API client https://github.com/octokit/rest.js ?

File: lib/utils/misc.js [hunk 2]
```diff
@@ -28,5 +28,17 @@ module.exports = {
   extractRepoName(url) {
     let pathItems = url.replace(/\/$/, '').split('/')
     return pathItems.pop().replace(/\.git$/, '')
+  },
+
+  extractGithubRepo(url) {
+    if (
```
Comment (xcambar) [inline lib/utils/misc.js]: `else { throw .. }` ?

File: lib/status/status.js (hunk 1 above)
Comment (njoyard) [reply lib/status/status.js]: Thanks for the pointer, I'll look into it. I just searched for "node github api" and took what looked the most sensible option :)

File: lib/utils/misc.js (hunk 2 above)
Comment (njoyard) [reply lib/utils/misc.js]: I chose not to throw when this is not a GH repo, as the behaviour of the user of this function (https://github.com/peopledoc/peon/pull/21/files#diff-846bea01946b5e6c06746e6e0429b58cR20) is to not push updates when it is not a GH repo. The idea being that we can watch/catch webhooks on non-GH repositories and there will just not be any status updates pushed to them.

File: lib/status/status.js (hunk 1 above)
Comment (njoyard) [reply lib/status/status.js]: Updated to use @octokit/rest

**Pull request merged**